src/bootstrap/setup.rs

```diff
@@ -107,6 +108,60 @@ pub fn setup(src_path: &Path, profile: Profile) {
     let include_path = profile.include_path(src_path);
     println!("`x.py` will now use the configuration at {}", include_path.display());
 
+    let build = TargetSelection::from_user(&env!("BUILD_TRIPLE"));
+    let stage_path = ["build", build.rustc_target_arg(), "stage1"].join("/");
+
+    let rustup_installed = Command::new("rustup")
+        .arg("--version")
+        .stdout(std::process::Stdio::null())
+        .output()
+        .map_or(false, |output| output.status.success());
+
+    let stage_dir_exists = match fs::create_dir(&stage_path[..]) {
+        Ok(_) => true,
+        Err(_) => Path::new(&stage_path[..]).exists(),
+    };
+
```

Comment:
IMO the current output is a little hard to read - maybe add a newline before any possible output?
```
Please choose one (a/b/c/d/e): a
`x.py` will now use the configuration at /home/joshua/src/rust/rust/src/bootstrap/defaults/config.library.toml
Added `stage1` rustup toolchain; try `cargo +stage1 build` on a separate rust project to run a newly-built toolchain

Rust's CI will automatically fail if it doesn't pass `tidy`, the internal tool for ensuring code quality.
```
```suggestion
println!();
```